syntax = "proto3";

message AIOSPacket {
    string session_id = 1;      
    uint64 seq_no = 2;           // Sequence number
    string data = 3;              // Data
    double ts = 4;               // Timestamp  
    string output_ptr = 5;       // JSON serialized string for output pointer
    repeated FileInfo files = 6; // Array of file structures
}

message FileInfo {
    string metadata = 1; // JSON serialized metadata
    bytes file_data = 2; // File data
}
